### PR TITLE
Consolidate prereq-install's

### PR DESCRIPTION
**Pull request by stephendade**
This is based off the discussion over at https://github.com/ArduPilot/ardupilot/issues/10717#issuecomment-470416788:
 - I've moved all the prereq-install-* into a single seperate folder
 - The Appveyor build now uses the prereq-install-windows script as a method of verifying that the script actually works.

Overall, makes it easier to check that the Windows dev environment still works. 

The plan is to edit the Wiki accordingly after this goes in.

### CODE REVIEW

**Comment by WickedShell** [comment]
@stephendade My understanding is that Appveyor is actually going away once some build sever scripts swap to the Azure build, so we might want to look at the Azure one instead. (I'm not fully up to speed on the plans there though)

**Comment by peterbarker** [comment]
Your first commit does more than just move the files.  It appears to introduce two new files!

**Comment by stephendade** [comment]
@peterbarker - appears that Git isn't tracking the moved files all that good. The two new files are derived from two deleted files, but they've been partially modified - apparently modified enough the Git's calling them "new" files rather than "moved and edited" files.

**Comment by peterbarker** [comment]
On Sun, 10 Mar 2019, Stephen wrote:

> @peterbarker - appears that Git isn't tracking the moved files all that good. The two new files are derived from two deleted files, but they've been
> partially modified - apparently modified enough the Git's calling them "new" files rather than "moved and edited" files.

Yes - generally you want to move - then have another commit which edits.

**Comment by stephendade** [comment]
@peterbarker - I've got back to this and re-done the commits to seperate out the move and edit changes.

I've left out the Appveyor changes for now. Need to figure out where Appveyor is going in the long term...

**Comment by peterbarker** [comment]
On Tue, 2 Apr 2019, Stephen wrote:

> @peterbarker - I've got back to this and re-done the commits to seperate out
> the move and edit changes.

Needs a rebase.

Peter

**Comment by peterbarker** [comment]
On Tue, 2 Apr 2019, Stephen wrote:

> @peterbarker - I've got back to this and re-done the commits to seperate out
> the move and edit changes.

... also, at some stage it would be good to get rid of "APM" whereever it 
lurks in this stuff.  Not necessarily here, 'though! :-)

Peter

File: Tools/environment_install/install-prereqs-windows-andAPMSource.ps1 [hunk 1]
```diff
@@ -4,25 +4,22 @@ Import-Module BitsTransfer
 
 Write-Output "Starting Downloads"
 
-Write-Output "Downloading MAVProxy (1/7)"
+Write-Output "Downloading MAVProxy (1/6)"
```
Comment (peterbarker) [inline Tools/environment_install/install-prereqs-windows-andAPMSource.ps1]: I'm *sure* powershell must have the concept a variable which could hold the number being changed on all these lines :-)

**Comment by stephendade** [comment]
Ok - is rebased now

**Comment by peterbarker** [comment]
This has probably broken the Vagrant install script - it calls the ubuntu install-prereqs script

Also... how do we handle updates to the Wiki?  There are going to be a lot of instructions on there that point to these scripts....

**Comment by stephendade** [comment]
Ok, Vagrant is fixed.

Wiki updates are in this PR: https://github.com/ArduPilot/ardupilot_wiki/pull/1692

**Comment by peterbarker** [comment]
```
pbarker@bluebottle:~/rc/ardupilot(master)$ git grep scripts/install
Dockerfile:    bash -c "Tools/scripts/install-prereqs-ubuntu.sh -y && apt-get install gcc-arm-none-eabi -y" && \
Tools/vagrant/initvagrant-trusty64.sh:/vagrant/Tools/scripts/install-prereqs-ubuntu.sh -y
Tools/vagrant/initvagrant-trusty64.sh:sudo -u $VAGRANT_USER /vagrant/Tools/scripts/install-jsbsim.sh /home/$VAGRANT_USER
Tools/vagrant/initvagrant.sh:sudo -H -u $VAGRANT_USER /vagrant/Tools/scripts/install-prereqs-ubuntu.sh -y
pbarker@bluebottle:~/rc/ardupilot(master)$ 
```

Docker is still broken, however.

**Comment by stephendade** [comment]
>Docker is still broken, however.

Fixed now

**Comment by peterbarker** [comment]
@stephendade I note you've removed the installation of JSBSim on Windows.  Are we going to add documentation on how to install it somewhere?

Also, we're trying to avoid using "APM" anywhere now.  We're renaming a script here to include "APM" in the name still, which isn't great.  Future enhancements welcome :-)

**Pull request merged**